### PR TITLE
Update to vite and fixes

### PR DESCRIPTION
**Pull request by qiandrewj**
# Summary

This PR updates CUReviews from react-scripts to vite, and makes a bunch of refactoring fixes!

## PR Type

- [ ] 🍕 Feature
- [X] 🐛 Bug Fix
- [X] 📝 Documentation Update
- [ ] 🎨 Style
- [X] 🧑‍💻 Code Refactor
- [X] 🔥 Performance Improvements
- [ ] ✅ Test
- [X] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## QA - Test Plan
Updating to vite is a big change, and from testing all of the aspects of the website, nothing has broken. It may require more investigation of packages that are not very compatible, and updating them if necessary. I will also be working on a test suite that is compatible with vite for a next PR.

Reporting reviews works, and the frontend is shown below:
Screenshot 2024-09-18 203114

## Breaking Changes & Notes
- Updated to vite. In the process, updated many package requirements to more recent versions. This has caused Github build to break, and the test suite is no longer compatible at all. **Will be working on addressing these issues next.** In order to test, you will have to delete node-modules and re-download packages.
- Refactored many functions to be async/await, and changed components to functional components. These have been tested and did not break any functionality.
- Updated all route names to be more clear and reflect best practices in endpoint naming style - also documented these changes for every route.
- Added report functionality back to each review with a simple button. This simply hides the review from the site and allows us to decide what to do with it.
- Other minor changes - fixing the profile dropdown, which wasn't toggling properly for me. Centered the loading icon! Finalizing adding/deleting admin users.    

## Added to documentation?

- [X] 📜 README.md
- [ ] 📓 notion
- [X] 🍕 Docstrings

### CODE REVIEW

File: client/src/modules/Admin/Components/Admin.tsx
Comment (jacquelinecai) [inline client/src/modules/Admin/Components/Admin.tsx:151]: Just tested this endpoint and it seems like the `semester` parameter is inaccurate. I don't think we ever set this semester to a value with the `setAddSemester` variable.

**Review by jacquelinecai — APPROVED** [review]
Thanks for making the update to vite and for all the code clean up work you did! This will really help strengthen our code quality and testing in the future!

I left one note about the admin buttons, but it should be a quick fix of setting the semester to update accordingly. Great work!

File: server/src/admin/admin.controller.ts
Comment (wizhaaa) [inline server/src/admin/admin.controller.ts:245]: Instead of "create" new admin user, maybe do "update user privilege" or something along those lines

File: server/src/ai/ai.router.ts
Comment (wizhaaa) [inline server/src/ai/ai.router.ts:1]: Make sure Helen knows about these changes too!

**Review by wizhaaa — CHANGES_REQUESTED** [review]
Hi Andrew! 

This is really great work that helps our codebase greatly. Thank you for putting in so much effort and time into updating packages, switching to Vite, adding documentation, renaming routes to more standard conventions, and refactoring other pieces of code. 

Vite is going to be super helpful in building our development environment quickly and helps to set us up to migrate away from Jest to Vitest too. 

I left a few comments on mainly stylistic choices (I only really looked into half the files closely since there were so many). I mainly just looked at the code and I did not see any major red flags/bugs/errors so far. 

There some minor merge conflicts that you may have to resolve still though. 

One note for the future, ideally it would be more practicable to chunk this PR into multiple smaller PRs with features more narrow in-scope. 

IE. 
- 1 Renaming routes and testing routes still work end to end 
- 1 for Report button 
- 1 for updating packages and into vite (probably have to refactor class modules to objects here too) 

Regardless, this is a great PR and I think everything should work! 

Hope to approve once the merge conflicts are resolved and figure out any other tasks blocking the CI pipeline.

**Pull request merged**
